packages/@aws-cdk/aws-cloudfront/lib/web_distribution.ts

```diff
@@ -622,6 +633,14 @@ export class CloudFrontWebDistribution extends cdk.Construct {
       };
     }
 
+    if (props.loggingConfig) {
+      distributionConfig.logging = {
+        bucket: (props.loggingConfig.bucket || new s3.Bucket(parent, `${name}LoggingS3Bucket`)).domainName,
```

Comment:
The `id` parameter can just be `"LogginBucket"`.

And please assign the resulting bucket (either the provided or generated one) to a public readonly member variable.

Reply:
Oh and does the bucket not need IAM permissions?

Reply:
Fixed the bucket name!

For the IAM:
Short: No
Long: CloudFront uses the root account it is attached to, that account needs permissions to change the ACL of the bucket. When a bucket is created inside an account it defaults to having these permissions. CloudFront uses that ACL permission to grant write perms to an internal amazon resource that collates the CloudFront logs into your one region.